### ISSUE TITLE
Build dev Docker image failed on CI

### ISSUE THREAD

**Issue by Yancey1989** [PaddlePaddle/Paddle#9073]
The develop Docker image building failed
CI log link: https://paddleci.ngrok.io/viewLog.html?buildId=30411&buildTypeId=Paddle_PrCi&tab=buildLog

And reproduce on host:

```bash
> docker build -t yancey1989/paddle:dev --build-arg UBUNTU_MIRROR=mirror://mirrors.ubuntu.com/mirrors.txt . --no-cache
....
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 zlib1g-dev : Depends: zlib1g (= 1:1.2.8.dfsg-2ubuntu4) but 1:1.2.8.dfsg-2ubuntu4.1 is to be installed
E: Unable to correct problems, you have held broken packages.
The command '/bin/sh -c apt-get update &&     apt-get install -y     git python-pip python-dev openssh-server bison     libnccl2=2.1.2-1+cuda8.0 libnccl-dev=2.1.2-1+cuda8.0     wget unzip unrar tar xz-utils bzip2 gzip coreutils ntp     curl sed grep graphviz libjpeg-dev zlib1g-dev      python-matplotlib gcc-4.8 g++-4.8     automake locales clang-format swig doxygen cmake      liblapack-dev liblapacke-dev     clang-3.8 llvm-3.8 libclang-3.8-dev     net-tools libtool &&     apt-get clean -y' returned a non-zero code: 100
...
```

**Comment by Yancey1989** [comment]
It works well when I remove the build argument: `UBUNTU_MIRROR=mirror://mirrors.ubuntu.com/mirrors.txt`, so I also remove it on CI to make the node work well.

**Comment by putcn** [comment]
interesting, looks `mirror://mirrors.ubuntu.com/mirrors.txt` should give more flexibility than just  `http://archive.ubuntu.com/ubuntu/` in apt's source list. I guess mirrors.ubuntu.com was temp down yesterday? let me test it out to see if we can add UBUNTU_MIRROR back.

**Comment by putcn** [comment]
Looks UBUNTU_MIRROR is working fine now. I added one configuration parameter in project PR_CI named `WITH_UBUNTU_MIRROR` to toggle UBUNTU_MIRROR ON and OFF. Currently I set this value to `ON`, if you see any further issue, try set it `OFF` in the temp build run dialog to see if that helps as a personal build. 
also, I found UBUNTU_MIRROR reduces the total build time a lot, I think we should keep it if possible.
